### PR TITLE
Elasticsearch integration

### PR DESCRIPTION
**Pull request by vaclavbohac**
Checkout the updated [readme](https://github.com/Demagog2/demagog/blob/2cbc8a84dd8d6c2775f1ac30d236ad959ee7a01d/README.md#elasticsearch-integration)

Canary release link https://canary.demagog.cz

TODOS:

* [x] design search result page (@vlki)
* [x] add search of segments (@vaclavbohac)
* [x] filter statements that are published for FE (search among unpublished could be still useful in admin) (@vaclavbohac)
* [x] search result paging (@vaclavbohac)
* [x] ~post commit hooks  (create, update, delete) – please mind soft deletes~
  * [x] ~articles~
  * [x] ~speakers~
  * [x] ~statements~
* [x] make elasticsearch work on circle
* [ ] setup capistrano tasks
* [x] ~add sidekiq to docker~

### CODE REVIEW

**Comment by vlki** [comment]
Relates to #184

**Comment by vaclavbohac** [comment]
> Want me to do the design of the search results page?

@vlki That would be really awesome man :-)

**Comment by vlki** [comment]
I checked analytics for the top searched terms:
```
okamura site:demagog.cz
babiš site:demagog.cz
zeman site:demagog.cz
Okamura site:demagog.cz
Babiš site:demagog.cz
kalousek site:demagog.cz
Zeman site:demagog.cz
bartoš site:demagog.cz
Kalousek site:demagog.cz
tomio okamura site:demagog.cz
Tomio Okamura site:demagog.cz
andrej babiš site:demagog.cz
tomio site:demagog.cz
Tomio site:demagog.cz
```

Looking at that, we should therefore prioritize speakers in the results and probably also the latest articles they were part of. Search among statements has the lowest priority on the results page.

Will try to put together the mockup today.

**Comment by vaclavbohac** [comment]
> I checked analytics for the top searched terms:
> 
> ```
> okamura site:demagog.cz
> babiš site:demagog.cz
> zeman site:demagog.cz
> Okamura site:demagog.cz
> Babiš site:demagog.cz
> kalousek site:demagog.cz
> Zeman site:demagog.cz
> bartoš site:demagog.cz
> Kalousek site:demagog.cz
> tomio okamura site:demagog.cz
> Tomio Okamura site:demagog.cz
> andrej babiš site:demagog.cz
> tomio site:demagog.cz
> Tomio site:demagog.cz
> ```
> 
> Looking at that, we should therefore prioritize speakers in the results and probably also the latest articles they were part of. Search among statements has the lowest priority on the results page.
> 
> Will try to put together the mockup today.

Great job man 👍

**Comment by vlki** [comment]
Here's the first mockup. In these the idea is that clicking the search icon in the top bar would bring the user to the search page with big input. User would type in the search term, click the search icon button in the input and that would bring up the results. First in the list would be speakers, then articles and finally statements. If there would be more than few speakers/articles/statements, it would show only the first few and then link to the all speakers/articles/statements.

search results - all
search resuts - articles only

Will give it for feedback to the team to find out if this can work.

What do you think @vaclavbohac ?

File: config/environments/development.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@
   config.active_record.migration_error = :page_load
 
   # Use local file system for active storage
-  config.active_storage.service = :local
+  config.active_storage.service = :amazon
```
Comment (vaclavbohac) [inline config/environments/development.rb]: Forgotten code

Comment (vaclavbohac) [reply config/environments/development.rb]:
```suggestion
  config.active_storage.service = :local
```

**Pull request merged**